p2p/protocol/identify/id.go

```diff
@@ -105,13 +110,15 @@ func NewIDService(ctx context.Context, h host.Host, opts ...Option) *IDService {
 		userAgent = cfg.userAgent
 	}
 
+	hostCtx, cancel := context.WithCancel(context.Background())
 	s := &IDService{
 		Host:      h,
 		UserAgent: userAgent,
 
-		ctx:           ctx,
+		ctx:           hostCtx,
+		ctxCancel:     cancel,
 		currid:        make(map[network.Conn]chan struct{}),
-		observedAddrs: NewObservedAddrSet(ctx),
+		observedAddrs: NewObservedAddrSet(hostCtx),
```

Comment:
@Stebalien 

Should we also have a `Close` for the `ObservedAddrSet` that we can call in `IDService.Close()`

Reply:
Some day, yes. But not today.